references/cli/status-ui/model/stack.go

```diff
@@ -0,0 +1,95 @@
+package model
+
+import "sync"
+
+const (
+	stackPush = iota
+	StackPop
+)
+
+type ResourceListener interface {
+	StackPop(Component, Component)
+	StackPush(Component)
+}
+
+type Stack struct {
+	components []Component
+	listeners  []ResourceListener
+	mx         sync.RWMutex
+}
+
+func NewStack() *Stack {
+	return &Stack{
+		components: make([]Component, 0),
+		listeners:  make([]ResourceListener, 0),
+	}
+}
+
+func (s *Stack) AddListener(listener ResourceListener) {
+	s.listeners = append(s.listeners, listener)
+}
+
+func (s *Stack) RemoveListener(listener ResourceListener) {
+	aimIndex := -1
+	for index, item := range s.listeners {
+		if item == listener {
+			aimIndex = index
+		}
+	}
+	if aimIndex == -1 {
+		return
+	}
+	s.listeners = append(s.listeners[:aimIndex], s.listeners[aimIndex+1:]...)
+}
+
+func (s *Stack) TopComponent() Component {
+	if s.Empty() {
+		return nil
+	}
+	component := s.components[len(s.components)-1]
```

Comment:
`return s.components[len(s.components)-1]`, No need add one local variable.

Reply:
ok, i have fixed it.